File: frontend/src/types/report.ts
/**
 * Report data types for frontend
 */

import type { DiscussionPhase } from '../../../shared/types'

/**
 * Report model
 */
export interface Report {
  id: string
  discussion_id: string
  overview: ReportOverview
  viewpoints_summary: ViewpointSummary[]
  consensus: Consensus
  controversies: Controversy[]
  insights: Insight[]
  recommendations: Recommendation[]
  quality_scores: QualityScores
  generation_time_ms: number
  created_at: string
}

export interface ReportOverview {
  topic: string
  participant_count: number
  duration_seconds: number
  round_count: number
  total_tokens: number
}

export interface ViewpointSummary {
  character_name: string
  character_stance: string
  main_arguments: string[]
  position_changes?: PositionChange[]
}

export interface PositionChange {
  from_phase: DiscussionPhase
  to_phase: DiscussionPhase
  old_position: string
  new_position: string
}

export interface Consensus {
  points: string[]
  supporting_arguments: Record<string, string[]>
}

export interface Controversy {
  topic: string
  disagreement_summary: string
  opposing_views: {
    character_name: string
    position: string
    arguments: string[]
  }[]
  resolution_status: 'unresolved' | 'partial' | 'resolved'
}

export interface Insight {
  category: string
  insight: string
  evidence: string[]
}

export interface Recommendation {
  priority: 'high' | 'medium' | 'low'
  recommendation: string
  rationale: string
}

export interface QualityScores {
  depth: number
  diversity: number
  constructive: number
  coherence: number
  overall: number
}

/**
 * Report export format
 */
export type ReportExportFormat = 'pdf' | 'markdown' | 'json'

File: backend/app/schemas/report.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, List, Dict, Any
from datetime import datetime
from uuid import UUID


class ReportOverview(BaseModel):
    topic_title: str
    discussion_duration_seconds: int
    total_rounds: int
    total_messages: int
    participant_count: int
    llm_provider: str
    llm_model: str


class ViewpointSummary(BaseModel):
    character_id: UUID
    character_name: str
    stance: str
    core_arguments: List[str]
    position_evolution: Optional[List[Dict[str, Any]]] = None


class Consensus(BaseModel):
    agreements: List[str]
    joint_recommendations: List[str]
    supporting_arguments: List[str]


class ControversyPoint(BaseModel):
    topic: str
    viewpoints: Dict[str, List[str]]  # {character_name: arguments}
    unresolved: bool


class Insight(BaseModel):
    description: str
    related_messages: List[UUID]


class Recommendation(BaseModel):
    action: str
    rationale: str
    priority: str


class QualityScores(BaseModel):
    depth: float  # 0-100
    diversity: float
    constructive: float
    coherence: float
    overall: float


class ReportResponse(BaseModel):
    id: UUID
    discussion_id: UUID
    overview: Optional[Dict[str, Any]] = None
    viewpoints_summary: Optional[List[Dict[str, Any]]] = None
    consensus: Optional[Dict[str, Any]] = None
    controversies: Optional[List[Dict[str, Any]]] = None
    insights: Optional[List[Dict[str, Any]]] = None
    recommendations: Optional[List[Dict[str, Any]]] = None
    quality_scores: Optional[Dict[str, float]] = None
    generation_time_ms: Optional[int] = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
